modules/tensor_mechanics/doc/content/documentation/systems/Materials/ComputePlaneFiniteStrain.md

```diff
@@ -1,14 +1,170 @@
+As in the classical presentation of the strain tensor in plane strain problems,
+the components of the deformation tensor associated with the $z$-direction are
+zero; these zero components indicate no coupling between the in-plane displacements
+and the out-of-plane strain variable.
+
+#### $X$-Direction of Out-of-Plane Strain
+
+If the user selects the out-of-plane direction as along the $x$-direction, the
+current and old deformation gradient tensors from [eqn:incremental_deformation_grad]
+are formulated as
+\begin{equation}
+  \label{eqn:deform_grads_xdirs}
+  \mathbf{A} = \begin{bmatrix}
+                F|^{dop} & 0 & 0 \\
+                0 & u_{y,y} & u_{z,y} \\
+                0 & u_{y,z} & u_{z,z}
+              \end{bmatrix} + \mathbf{I}
+  \qquad \text{  and  } \qquad
+  \bar{\mathbf{F}} = \begin{bmatrix}
+                F|^{dop}_{old} & 0 & 0 \\
+                0 & u_{y,y}|_{old} & u_{y,z}|_{old} \\
+                0 & u_{z,y}|_{old}& u_{z,z}|_{old}
+              \end{bmatrix} + \mathbf{I}
+\end{equation}
+so that the off-diagonal components of the deformation tensors associated with
+the $x$-direction are zeros.
+
+#### $Y$-Direction of Out-of-Plane Strain
+
+If the user selects the out-of-plane direction as along the $y$-direction, the
+current and old deformation gradient tensors from [eqn:incremental_deformation_grad]
+are formulated as
+\begin{equation}
+  \label{eqn:deform_grads_ydirs}
+  \mathbf{A} = \begin{bmatrix}
+                u_{x,x} & 0 & u_{z,x} \\
+                0 & F|^{dop} & 0 \\
+                u_{x,z} & 0 & u_{z,z}
+              \end{bmatrix} + \mathbf{I}
+  \qquad \text{  and  } \qquad
+  \bar{\mathbf{F}} = \begin{bmatrix}
+                u_{x,x}|_{old} & 0 & u_{x,z}|_{old} \\
+                0 & F|^{dop}_{old} & 0 \\
+                u_{z,x}|_{old} & 0 & u_{z,z}|_{old}
+              \end{bmatrix} + \mathbf{I}
+\end{equation}
+so that the off-diagonal components of the deformation tensors associated with
+the $y$-direction are zeros.
+
+
+### Finalized Deformation Gradient
+
+If selected by the user, the incremental deformation gradient is conditioned with
+a $\bar{B}$ formulation to mitigate volumetric locking of the elements.
+The volumetric locking correction is applied to both the incremental deformation
+gradient
+\begin{equation}
+  \label{eqn:vlc_fhat}
+  \hat{\mathbf{F}}|_{vlc} = \left( \frac{1}{det(\hat{\mathbf{F}})} \frac{\hat{\mathbf{F}}_{avg}}{V_{elem}} \right)^{1/3}
+\end{equation}
+and the total deformation gradient. For more details about the theory behind
+[eqn:vlc_fhat] see the
+[Volumetric Locking Correction](/tensor_mechanics/VolumetricLocking.md)
+documentation.
+
+Once the incremental deformation gradient is calculated for the specific 2D geometry,
+the deformation gradient is passed to the strain and rotation methods used by the
+3D Cartesian simulations, as described in the [Finite Strain Class](ComputeFiniteStrain.md)
+documentation.
+
+## Example Input File
```

Comment:
It might be good to also include an example of regular plane strain here as well with the Master Action as the `planar_formulation` is then simply `PLANE_STRAIN` and the `scalar_out_of_plane_strain` parameter is not needed.   This could also be good because the GPS examples are included in the GPS documentation.  

Maybe if one is added for regular plane strain you could use one of the examples that has the `out_of_plane_direction` not the default.

Reply:
Now that these tests are in MOOSE I can do that. Would you point me to a test that you had in mind?

Reply:
tensor_mechanics/test/tests/2D_different_planes/planestrain_xz.i should work.

Reply:
Could only add this for the small strain case because there are not any finite strain cases checked in: I can't pull in example syntax from the `[Tests]` cli_args